### PR TITLE
feat: handle commands on the client side

### LINKED ISSUE #49: handle commands on the client side
Some commands on the `ltex` lsp are resolved on the client side.

Example:

- `_ltex.addToDictionary`
- `_ltex.disableRules`
- `_ltex.hideFalsePositives`

[Docs: ltex commands](https://valentjn.github.io/ltex/ltex-ls/server-usage.html#commands)

```lua
local lspconfig = require 'lspconfig'
local opts = {
  root_dir = function(filename)
    return lspconfig.util.path.dirname(filename)
  end,
  on_init = function(client, _)
    vim.lsp.commands['_ltex.addToDictionary'] = function(cmd, ctx)
      print(vim.inspect { cmd, ctx })
    end
  end,
  settings = {
    ltex = {
      completionEnabled = true,
      language = 'en-US',
      checkFrequency = 'edit',
      setenceCacheSize = 5000,
      additionalRules = {
        enablePickyRules = true,
        motherTongue = 'en-US',
      },
      dictionary = {
        ['en-US'] = { 'Hello', 'Neovim' },
      },
    },
  },
}
lspconfig.ltex.setup(opts)
```

Using `:CodeActionMenu` not run the command `_ltex.addToDictionary`. But using `:lua vim.lsp.buf.code_action()` works.

https://user-images.githubusercontent.com/16160544/160704020-a61c2bf6-e8be-4524-b2fb-eafd4d90c71e.mp4

### PR DESCRIPTION
**Pull request by aspeddro**
### Main Change

- `BaseAction:new` now takes `client_id` in the second index of the table

Feel free to make suggestions.

Close #49

### CODE REVIEW

File: lua/code_action_menu/lsp_objects/actions/base_action.lua [hunk 1]
```diff
@@ -4,10 +4,9 @@ local WorkspaceEdit = require(
 
 local BaseAction = {}
 
-function BaseAction:new(server_data)
-  vim.validate({ ['server data'] = { server_data, 'table' } })
-
-  local instance = { server_data = server_data }
+function BaseAction:new(data)
+  vim.validate({ ['data'] = { data, 'table' } })
+  local instance = { server_data = data[1], client_id = data[2] }
```
Comment (weilbith) [inline lua/code_action_menu/lsp_objects/actions/base_action.lua:9]: Why did you rename this here. Is it somehow related to the feature? 🤔

File: lua/code_action_menu/lsp_objects/actions/code_action.lua [hunk 2]
```diff
@@ -80,7 +80,24 @@ function CodeAction:execute()
   if self:is_workspace_edit() then
     vim.lsp.util.apply_workspace_edit(self.server_data.edit, 'utf-8')
   elseif self:is_command() then
-    vim.lsp.buf.execute_command(self.server_data.command)
+    local client = vim.lsp.get_client_by_id(self.client_id)
+    local fn = client.commands[self.server_data.command.command]
```
Comment (weilbith) [inline lua/code_action_menu/lsp_objects/actions/code_action.lua:84]: In my tests, this is not correct and also not how the native implementation does it. According to the specification this is just `self.server_data.command`. Do you have a language server which formats that differently?

File: lua/code_action_menu/lsp_objects/actions/code_action.lua [hunk 3]
```diff
@@ -80,7 +80,24 @@ function CodeAction:execute()
...
+      or vim.lsp.commands[self.server_data.command.command]
+    if fn then
+      local context = {}
+      context.diagnostic = vim.lsp.diagnostic.get_line_diagnostics()
+      local params = vim.lsp.util.make_range_params()
+      params.context = context
+
+      fn(self.server_data.command, {
```
Comment (weilbith) [inline lua/code_action_menu/lsp_objects/actions/code_action.lua:92]: Related to the above comment: we just need to pass `self.server_data` here. The `self.server_data.command` is just the `string`.

**Review by weilbith — CHANGES_REQUESTED** [review]
Thanks for implementing this feature! 🙏🏾 

It is not yet working for me. But I think that can be reduced to one simple issue (see comments).
The second point is that you added it for workspace edits which are detected to be actually a command (legacy language servers), but for the actual `Command` class you did not add it. Could you please do so? Don't worry about duplicated code here. Will be resolved in future anyway. 🙂

File: lua/code_action_menu/lsp_objects/actions/code_action.lua (hunk 3 above)
Comment (aspeddro) [reply lua/code_action_menu/lsp_objects/actions/code_action.lua:92]: Nope, `self.sever_data.command` is a table.

Add the following at line 91

```lua
print(vim.inspect('here --> ' .. type(self.server_data.command)))
```

See the output https://github.com/weilbith/nvim-code-action-menu/issues/49#issue-1185397324

File: lua/code_action_menu/lsp_objects/actions/code_action.lua (hunk 2 above)
Comment (aspeddro) [reply lua/code_action_menu/lsp_objects/actions/code_action.lua:84]: Can you share your tests?

The navite implementation:

```lua
local fn = client.commands[command.command] or vim.lsp.commands[command.command]
```

https://github.com/neovim/neovim/blob/75ff156d9b11f23f6156cd6b90fb7bad0a83fd18/runtime/lua/vim/lsp/buf.lua#L512-L513
